pkg/core/managers/apis/external_service/external_service_validator.go

```diff
@@ -0,0 +1,64 @@
+package externalservice
+
+import (
+	"context"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	"github.com/kumahq/kuma/pkg/core/validators"
+)
+
+type ExternalServiceValidator struct {
+	Store store.ResourceStore
+}
+
+func (r *ExternalServiceValidator) ValidateCreate(ctx context.Context, mesh string, resource *core_mesh.ExternalServiceResource) error {
+	return r.validateRateLimits(ctx, mesh, resource.Spec.GetTags()[mesh_proto.ServiceTag])
+}
+
+func (r *ExternalServiceValidator) ValidateUpdate(ctx context.Context, previousExternalService *core_mesh.ExternalServiceResource, newExternalService *core_mesh.ExternalServiceResource) error {
+	return r.validateRateLimits(ctx, previousExternalService.GetMeta().GetMesh(), newExternalService.Spec.GetTags()[mesh_proto.ServiceTag])
+}
+
+func (r *ExternalServiceValidator) ValidateDelete(ctx context.Context, name string) error {
+	return nil
+}
+
+func (r *ExternalServiceValidator) validateRateLimits(ctx context.Context, mesh string, service string) error {
+	validationErr := &validators.ValidationError{}
+	// If referenced by a ratelimit, that ratelimit must only match kuma.io/service destinations
+
+	rateLimits := &core_mesh.RateLimitResourceList{}
+	err := r.Store.List(ctx, rateLimits, store.ListByMesh(mesh))
+	if err != nil {
+		if store.IsResourceNotFound(err) {
+			return nil
+		} else {
+			validationErr.AddViolation("externalservice", err.Error())
+			return validationErr
+		}
+	}
+
+	for _, rateLimit := range rateLimits.Items {
+		for _, dest := range rateLimit.Destinations() {
+			matchesThisService := false
+			matchesNonService := false
+			for tag, value := range dest.GetMatch() {
+				if tag == mesh_proto.ServiceTag {
+					if value == service {
```

Comment:
what about `*`?

why are we iterating over all pairs?

I don't think this algo correctly detects what we want to detect.
You need to check whether the ExternalService matches RateLimit and if so, is there a match against more than `kuma.io/service`

Here are a couple of examples
1) Pass
```
type: ExternalService
tags:
  kuma.io/service: xyz
---
type: RateLimit
destinations:
- match:
    kuma.io/service: xyz
```

2) Pass
```
type: ExternalService
tags:
  kuma.io/service: xyz
---
type: RateLimit
destinations:
- match:
    kuma.io/service: '*'
```

3) Fail because of match on other tags than kuma.io/service: xyz
```
type: ExternalService
tags:
  kuma.io/service: xyz
  version: v1
---
type: RateLimit
destinations:
- match:
    kuma.io/service: xyz
    version: v1
```

4) Pass because it does not match
```
type: ExternalService
tags:
  kuma.io/service: xyz
  version: v1
---
type: RateLimit
destinations:
- match:
    kuma.io/service: xyz
    version: v2
```

Reply:
After some more discussion we decided that Example #4, above should Fail - Validation rule is "Any RateLimit which includes a kuma.io/service tag matching an external service should not include any other tags".